Tasks/PowerShellV2/powershell.ps1

```diff
@@ -74,11 +74,22 @@ try {
     #
     # Note, use "-Command" instead of "-File". On PowerShell v4 and V3 when using "-File", terminating
     # errors do not cause a non-zero exit code.
+    #
+    # Adds in logic to try and use pwsh.exe first over powershell if installed on the agent.   
+    #
+    if ((Get-Command -name pwsh.exe -CommandType Application -ErrorAction SilentlyContinue).count -gt 0) 
+                    { $pwshPath = Get-Command -name pwsh.exe -CommandType Application | Select-Object -First 1 -ExpandProperty Path }
```

Comment:
The output of `Get-Command` should be stored in a temporary variable to not execute the command twice.